src/service/resources-impl.js

```diff
@@ -407,13 +410,29 @@ export class Resources {
    * @private
    */
   buildReadyResources_() {
+    // Avoid cases where elements add more elements inside of them
+    // and cause an infinite loop of building - see #3354 for details.
+    if (this.isCurrentlyBuildingPendingResources_) {
```

Comment:
That shouldn't be the case since the [for-loop](https://github.com/ampproject/amphtml/pull/3355/files#diff-80b9cea934a00a64f15d14cb656dfad4R428) will not stop until everything that's in the queue (even things added after the for-loop starts, because we're doing `i < .pending.length` not `i < cachedLengthVar`) has been iterated over.